speaker/main.go

```diff
@@ -314,6 +326,12 @@ func (c *controller) SetConfig(l log.Logger, cfg *config.Config) k8s.SyncState {
 }
 
 func (c *controller) SetNode(l log.Logger, node *v1.Node) k8s.SyncState {
+	nodeLabels := node.Labels
+	if nodeLabels == nil {
+		nodeLabels = map[string]string{}
+	}
+	ns := labels.Set(nodeLabels)
+	c.nodeLabels = ns
```

Comment:
It looks like this is going to conflict with code in `speaker/bgp_controller.go`.  Take a look at `SetNode()` in there.